src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -764,6 +766,11 @@ export abstract class BaseExtHostTerminalService extends Disposable implements I
 
 		return completions;
 	}
+	// Take in shellType as a string and return VSCode Terminal Shell Type?
+	public async $acceptTerminalShellType(id: number, shellType: string): Promise<TerminalShellType> {
+		// TODO: Implement
+		return GeneralShellType.Python;
+	}
```

Comment:
Look at this for something similar to what you need to do:

https://github.com/microsoft/vscode/blob/e8184ed571509b6bc1f7752ff2216116f6194ff3/src/vs/workbench/api/common/extHostTerminalService.ts#L686-L691

Basically get the terminal for the ID, set it on it (interacted with is also on the `vscode.TerminalState` object), then fire the `onDidChangeTerminalState` event.